src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,123 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @Test
+    void userAdd_correctUser_Ok() {
+        User user = new User(123L, "correctLogin", "correctPassword", 20);
+        User actual = registrationService.register(user);
+        assertEquals(User.class, actual.getClass());
+    }
+
+    @Test
+    void nullUser_null_notOk() {
+        User user = null;
+        assertThrows(ValidationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void sameUser_repeatLogin_notOK() {
+        User user = new User(123L, "repeatLogin", "correctPassword", 18);
+        Storage.people.add(user);
+        assertThrows(ValidationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void loginLength_3Length_notOk() {
+        User user = new User(123L, "sho", "correctPassword", 20);
+        assertThrows(ValidationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void loginLength_5Length_notOk() {
+        User user = new User(123L, "short", "correctPassword", 20);
+        assertThrows(ValidationException.class, () -> registrationService.register(user));
```

Comment:
```suggestion
        assertThrows(ValidationException.class, () -> registrationService.register(user), "If user login is short it should throw ValidationException Exception!");
```

Reply:
@jszymanski97 a message for the violation of assertion

Reply:
@jszymanski97 Personally I think such message is redundant here - the reason for throwing an exception is provided in a test method's name.

IMO your tests could be improved by evaluating whether the message of thrown exception matches the expected value from `ErrorMessage` class for each case.

Reply:
Okay, Thank you I think I get the point now… Checking if correct message is returned instead correct exception. Will get these corrected later on today
Thank You